### PR TITLE
Return the full redirect, including URL prefix, if any

### PR DESCRIPTION
**Pull request by christophermaier**
The executor needs the prefix in order to determine which adapter to use.

### CODE REVIEW

**Comment by kevsmith** [comment]
Looks good overall. The only question I have is whether we should standardize on the string representation of a redirect. As it stands with this PR Piper can return two different strings naming the same redirect target:

```
iex(8)> {:ok, ast} = Piper.Command.Parser.scan_and_parse("foo | bar | baz > chat://#room1")
{:ok, ...}
iex(9)> Piper.Command.Ast.Pipeline.redirect_targets(ast)
["chat://#room1"]
```
```
iex(10)> {:ok, ast} = Piper.Command.Parser.scan_and_parse("foo | bar | baz > #room1")
{:ok, ...}
iex(11)> Piper.Command.Ast.Pipeline.redirect_targets(ast)
["#room1"]
```
I think it makes sense to standardize on the first notation -- meaning Piper would insert `chat://` where needed -- but I have no idea how much work it would be to implement.

**Comment by christophermaier** [comment]
@kevsmith I think what you illustrated is the desired behavior. The executor treats them as potentially different. That is `#room1` is a destination handled by the _originating adapter_, whereas `chat://#room1` is a destination handled by the currently configured _chat_ adapter, which may not be the same (as is the case for HTTP-initiated pipelines).

While this is not a big deal for redirecting to rooms, it  would require Piper to be aware of our "special" redirects so it wouldn't, e.g., turn `here` into `chat://here` which would totally change the meaning. And since we don't impose a uniform syntax on room / user names (opting to obey the convention of the chat provider), we'd cause issues for people whose platform conventions and names cause conflicts with our redirects (the pathological case of someone having a HipChat room named `here`, for instance). One nice thing about the `chat://` prefix is that it allows users to specifically override things in such cases.

**Comment by kevsmith** [comment]
The point I was trying to make was that we should have a consistent internal notation for redirect destinations -- chat targets should always be prefixed with `chat://`, web targets (when we have them) should always be prefixed with `web://`. 

If this isn't possible now, and it sounds like it's not, then I'm fine with punting the idea to sometime in the future.

tl;dr :+1:

**Comment by christophermaier** [comment]
Yeah, we'd need to feed Piper more information for this to work, but consistency sounds good. I'll make an issue to revisit this. Thanks.
